File: solutions/First-Missing-Positive/First-Missing-Positive.cpp

class Solution {
public:
    int firstMissingPositive(int A[], int n) {
        for(int i=0;i<n;i++)
            if(A[i]>0 && A[i]<n)
                if(A[i]-1!=i && A[i]!=A[A[i]-1])
                {
                    swap(A[i],A[A[i]-1]);
                    i--;
                }
        for(int i=0;i<n;i++)
            if(A[i]-1!=i)
                return i+1;
        return n+1;
    }
};


 

File: solutions/First-Missing-Positive/First-Missing-Positive.py


class Solution:
    # @param A, a list of integers
    # @return an integer
    def firstMissingPositive(self, A):
        if len(A)==0:
            return 1
        hash=[0 for i in range(10000)]
        for i in range(len(A)):
            if A[i]>0:
                hash[A[i]]=1
        for i in range(1,10000):
            if hash[i]==0:
                return i
 